tools/aws-ami-cleanup.sh

```diff
@@ -0,0 +1,148 @@
+#!/bin/bash
+set -euo pipefail
+
+# Set up variables.
+# Maximum idle timeout in hours
+IDLE_TO=4
+UPSTREAM_PREFIX_NAME="composer-ci-"
+DOWNSTREAM_PREFIX_NAME="rhel-edge-"
+BUCKET_LIST=bucket_list.json
+SNAPSHOT_LIST=snapshot_list.json
+IMAGES_LIST=images_list.json
+KEYPAIR_LIST=keypair_list.json
+
+# Colorful output.
+function greenprint {
+    echo -e "\033[1;32m${1}\033[0m"
+}
+
+# Calculate how old is the resource
+time_subtraction () {
+    current_time=$(date +%s)
+    creation_time=$1
+    
+    hours_old_float=$(
+        echo "${current_time}" \
+            $(date -d "${creation_time}" +%s) \
+            | awk '{print ($1 - $2) / 3600}'
+    )
+    hours_old_int=$(printf "%.0f\n" "${hours_old_float}")
+    if [[ "${hours_old_int}" -ge "${IDLE_TO}" ]]; then
+        echo 1
+    else
+        echo 0
+    fi
+}
+
+greenprint "🧼 Cleaning up AWS resources in region ${AWS_DEFAULT_REGION}"
+
+# List Upstream and Downstream Buckets
+greenprint "Checking AWS S3 Idle Buckets 🪣"
+aws s3 ls \
+    | grep -e "${UPSTREAM_PREFIX_NAME}" \
+    -e "${DOWNSTREAM_PREFIX_NAME}" \
+    > "${BUCKET_LIST}"
+if [[ -z $(cat "${BUCKET_LIST}") ]]; then
+    echo "No buckets to remove"
+else
+    mapfile -t BUCKET_ARRAY < "${BUCKET_LIST}"
+    rm "${BUCKET_LIST}"
+    
+    for line in "${BUCKET_ARRAY[@]}"; do
+        CREATION_DATE=$(
+            echo "${line}" \
+                | awk -F " |" '{print ($1" "$2)}'
+        )
+        BUCKET_NAME=$(
+            echo "${line}" \
+                | awk -F " |" '{print ($3)}'
+        )
+        BUCKET_AGE=$(time_subtraction "${CREATION_DATE}")
+        if [[ "${BUCKET_AGE}" == 1 ]]; then
+           echo "Removing bucket ${line}"
+           aws s3 rb "s3://${BUCKET_NAME}" --force > /dev/null
+        fi
+    done
+fi
+
+# List Registered Images
+greenprint "Checking AWS EC2 Idle AMIs 💽"
+aws ec2 describe-images \
+    --owners self \
+    --query 'Images[?(Tags[?Value == `composer-ci`].Value)]' \
```

Comment:
Is the `'` required? Can it be changed to `"`? Thanks.

Reply:
If we decide to keep single quotes, then we will need to land [PR#4210](https://github.com/virt-s1/rhel-edge/pull/4210)

Reply:
Another workaround has been added. It is just to ignore Shellcheck SC2016.